reference/test/reorder/mc64_kernels.cpp

```diff
@@ -0,0 +1,401 @@
+    using matrix_type = typename TestFixture::matrix_type;
+    using permutation_type = typename TestFixture::permutation_type;
+
+    auto mc64_factory =
+        gko::reorder::Mc64<value_type, index_type>::build()
+            .with_strategy(gko::reorder::mc64_strategy::max_diagonal_product)
+            .on(this->ref);
+    auto mc64 = mc64_factory->generate(this->mtx);
+
+    auto perm = mc64->get_permutation()->get_const_permutation();
+    auto inv_perm = mc64->get_inverse_permutation()->get_const_permutation();
+    auto row_scaling = mc64->get_row_scaling()->get_const_values();
+    auto col_scaling = mc64->get_col_scaling()->get_const_values();
+
+    GKO_ASSERT_EQ(perm[0], 1);
+    GKO_ASSERT_EQ(perm[1], 5);
+    GKO_ASSERT_EQ(perm[2], 3);
+    GKO_ASSERT_EQ(perm[3], 4);
+    GKO_ASSERT_EQ(perm[4], 0);
+    GKO_ASSERT_EQ(perm[5], 2);
+    GKO_ASSERT_EQ(inv_perm[0], 4);
+    GKO_ASSERT_EQ(inv_perm[1], 0);
+    GKO_ASSERT_EQ(inv_perm[2], 5);
+    GKO_ASSERT_EQ(inv_perm[3], 2);
+    GKO_ASSERT_EQ(inv_perm[4], 3);
+    GKO_ASSERT_EQ(inv_perm[5], 1);
+    GKO_ASSERT_NEAR(row_scaling[0], value_type{1. / 3.}, this->tolerance);
+    GKO_ASSERT_NEAR(row_scaling[1], value_type{0.2}, this->tolerance);
+    GKO_ASSERT_NEAR(row_scaling[2], value_type{0.2}, this->tolerance);
+    GKO_ASSERT_NEAR(row_scaling[3], value_type{4. / 15.}, this->tolerance);
+    GKO_ASSERT_NEAR(row_scaling[4], value_type{0.3}, this->tolerance);
+    GKO_ASSERT_NEAR(row_scaling[5], value_type{2. / 15.}, this->tolerance);
+    GKO_ASSERT_NEAR(col_scaling[0], value_type{1.}, this->tolerance);
+    GKO_ASSERT_NEAR(col_scaling[1], value_type{1.5}, this->tolerance);
+    GKO_ASSERT_NEAR(col_scaling[2], value_type{0.9375}, this->tolerance);
+    GKO_ASSERT_NEAR(col_scaling[3], value_type{5. / 6.}, this->tolerance);
+    GKO_ASSERT_NEAR(col_scaling[4], value_type{1.}, this->tolerance);
+    GKO_ASSERT_NEAR(col_scaling[5], value_type{1.25}, this->tolerance);
+}
+
+
+TYPED_TEST(Mc64, CreatesCorrectPermutationAndScalingLargeTrivialExampleProduct)
+{
+    using index_type = typename TestFixture::index_type;
+    using real_type = typename TestFixture::real_type;
+    using value_type = typename TestFixture::value_type;
+    using matrix_type = typename TestFixture::matrix_type;
+
+    std::ifstream mtx_stream{gko::matrices::location_1138_bus_mtx};
+    auto mtx = gko::share(gko::read<matrix_type>(mtx_stream, this->ref));
+    std::ifstream result_stream{gko::matrices::location_1138_bus_mc64_result};
+    auto expected_result = gko::read<matrix_type>(result_stream, this->ref);
+
+    auto mc64_factory =
+        gko::reorder::Mc64<value_type, index_type>::build()
+            .with_strategy(gko::reorder::mc64_strategy::max_diagonal_product)
+            .on(this->ref);
+    auto mc64 = mc64_factory->generate(mtx);
+
+    auto perm = mc64->get_permutation();
+    auto row_scaling = mc64->get_row_scaling();
+    auto col_scaling = mc64->get_col_scaling();
```

Comment:
```suggestion
    std::ifstream mtx_stream{gko::matrices::location_1138_bus_mtx};
    auto mtx = gko::share(gko::read<matrix_type>(mtx_stream, this->ref));
    std::ifstream result_stream{gko::matrices::location_1138_bus_mc64_result};
    auto expected_result = gko::read<matrix_type>(result_stream, this->ref);
    auto mc64_factory =
        gko::reorder::Mc64<value_type, index_type>::build()
            .with_strategy(gko::reorder::mc64_strategy::max_diagonal_product)
            .on(this->ref);
    auto mc64 = mc64_factory->generate(mtx);
    auto perm = mc64->get_permutation();
    auto row_scaling = mc64->get_row_scaling();
    auto col_scaling = mc64->get_col_scaling();
```
or add some comment to separate them.